src/followup_questions/followup_questions.py

```diff
@@ -0,0 +1,92 @@
+import typing
+from typing import Protocol
+
+import tiktoken
+from langchain_core.messages import BaseMessage
+from langchain_core.messages.utils import get_buffer_string
+from langchain_core.prompts import PromptTemplate
+
+from agents.common.utils import filter_messages
+from followup_questions.constants import (
+    DEFAULT_HISTORY_MESSAGES_LIMIT,
+    DEFAULT_TOKEN_LIMIT,
+)
+from followup_questions.prompts import FOLLOW_UP_QUESTIONS_PROMPT
+from initial_questions.inital_questions import IEncoding
+from initial_questions.output_parser import QuestionOutputParser
+from utils.logging import get_logger
+from utils.models import IModel
+
+logger = get_logger(__name__)
+
+
+class IFollowUpQuestionsHandler(Protocol):
+    """Protocol for IFollowUpQuestionsHandler."""
+
+    def generate_questions(self, messages: list[BaseMessage]) -> list[str]:
+        """Generates initial questions given a context with cluster data."""
```

Comment:
Is this used for initial q's as well?

Reply:
no, only for followup questions. Initila questions have a separate method with similar name.

Reply:
It says "initial questions" right now

Reply:
Good catch! Updated!